pkg/pillar/rootfs/init.sh

```diff
@@ -3,7 +3,9 @@
 # This *really* needs to be replaced with tini+monit ASAP.
 
 # Start with a default content for resolv.conf
-echo 'nameserver 8.8.8.8' > /etc/resolv.conf
+echo 'nameserver 127.0.0.1' > /etc/resolv.conf
```

Comment:
Maybe instead of rewriting `/etc/resolv.conf` and enforcing the use of the cache for all, I would suggest to apply the caching only for datastore downloaders here: https://github.com/lf-edge/eve/blob/3ea90200948f2723f1123cf2ed3565c4e2012bef/libs/zedUpload/datastore.go#L78
Inside the resolver you can override the address with `127.0.0.1:53`. This is a trick mentioned for example [here](https://koraygocmen.medium.com/custom-dns-resolver-for-the-default-http-client-in-go-a1420db38a5d).
This would give us more control over DNS and we can decide for which requests to use the cache (e.g. maybe only for some specific datastores).

Reply:
Good point, modified